frontend/src/components/data-table/chart-transforms/form-components.tsx

```diff
@@ -1,89 +1,133 @@
 /* Copyright 2024 Marimo. All rights reserved. */
 
+import React from "react";
+import { capitalize } from "lodash-es";
+import { XIcon, PlusIcon, SquareFunctionIcon } from "lucide-react";
+import type { UseFormReturn, Path, PathValue } from "react-hook-form";
+import type { z } from "zod";
+
+import type { DataType } from "@/core/kernel/messages";
+import type { NumberFieldProps } from "@/components/ui/number-field";
+import type { ChartSchema, ScaleType } from "./chart-schemas";
+
 import {
   FormField,
   FormItem,
   FormLabel,
   FormControl,
 } from "@/components/ui/form";
-import type { UseFormReturn, Path, PathValue } from "react-hook-form";
-import type { DataType } from "@/core/kernel/messages";
 import {
   Select,
   SelectContent,
   SelectGroup,
   SelectItem,
+  SelectLabel,
   SelectTrigger,
   SelectValue,
 } from "@/components/ui/select";
 import { DATA_TYPE_ICON } from "@/components/datasets/icons";
 import { DebouncedInput, DebouncedNumberInput } from "@/components/ui/input";
-import { SquareFunctionIcon } from "lucide-react";
 import { Checkbox } from "@/components/ui/checkbox";
-import { cn } from "@/utils/cn";
-import { DEFAULT_BIN_VALUE, NONE_GROUP_BY } from "./chart-schemas";
-import type { NumberFieldProps } from "@/components/ui/number-field";
 import { Button } from "@/components/ui/button";
-import { XIcon, PlusIcon } from "lucide-react";
-import React from "react";
+import { cn } from "@/utils/cn";
+import { SliderComponent } from "@/plugins/impl/SliderPlugin";
+import { Multiselect } from "@/plugins/impl/MultiselectPlugin";
```

Comment:
im a bit wary of this file depending on these plugin components, in case styling and props drift over time. 

could we create a share component or sub-components?